tests/test_migrations.py

```diff
@@ -42,88 +85,138 @@ async def test_4d93784a25e5_downgrade(
     alembic_main(["--raiseerr", "upgrade", "4d93784a25e5"])
 
     # data values
-    fake_guid = "7891011"
+    fake_guid = "123456"
 
     async with db.with_bind(DB_DSN):
         # insert data
+        sql_authz_data = escape(json.dumps(authz_data))
         if new_metadata is not None:
-            insert_stmt = f"INSERT INTO metadata(\"guid\", \"authz\", \"data\") VALUES ('{fake_guid}', '{authz_data}', '{new_metadata}')"
+            sql_new_metadata = escape(json.dumps(new_metadata))
+            insert_stmt = f"INSERT INTO metadata(\"guid\", \"authz\", \"data\") VALUES ('{fake_guid}', '{sql_authz_data}', '{sql_new_metadata}')"
         else:
-            insert_stmt = f'INSERT INTO metadata("guid", "authz", "data") VALUES (\'{fake_guid}\', \'{authz_data}\', null)'
-            # null will come back as 'None' in select statement
-            new_metadata = "None"
-        data = await db.scalar(db.text(insert_stmt))
+            insert_stmt = f'INSERT INTO metadata("guid", "authz", "data") VALUES (\'{fake_guid}\', \'{sql_authz_data}\', null)'
+        await db.scalar(db.text(insert_stmt))
 
         # check that the request data was inserted correctly
-        data = await db.all(db.text(f"SELECT * FROM metadata"))
-        request = {k: str(v) for row in data for k, v in row.items()}
-        assert request["guid"] == fake_guid
-        assert request["authz"] == authz_data.replace('"', "'")
-        assert request["data"] == new_metadata.replace('"', "'")
+        data = await db.all(
+            db.text(f"SELECT * FROM metadata WHERE guid = '{fake_guid}'")
+        )
+        row = {k: v for k, v in data[0].items()}
+        assert row == {"guid": fake_guid, "data": new_metadata, "authz": authz_data}
 
         # downgrade to before "add_authz_column" migration
         alembic_main(["--raiseerr", "downgrade", "f96cb3b2c523"])
 
-        # check that the migration moved the `authz` data into the `data` column
-        data = await db.all(db.text("SELECT * FROM metadata"))
-        request = {k: v for row in data for k, v in row.items()}
-        assert request["guid"] == fake_guid
-        assert "authz" not in request
-        assert request["data"] == old_metadata
+        try:
```

Comment:
the `try/finally` block should include all the statements starting from after inserting data